Jantimizer/Histograms/Managers/MinDepthHistogramManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Histograms.Caches;
+using Histograms.DataGatherers;
+using Histograms.HistogramSelectors;
+using Histograms.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Tools.Caches;
+using Tools.Models;
+
+namespace Histograms.Managers
+{
+    public class MinDepthHistogramManager : BaseHistogramManager, IDepthHistogramManager
+    {
+        public int Depth { get; }
+
+        public MinDepthHistogramManager(IDataGatherer dataGatherer, int depth) : base(dataGatherer)
+        {
+            Depth = depth;
+        }
+
+        protected override async Task<IHistogram> CreateHistogramForAttribute(string tableName, string attributeName)
+        {
+            IDepthHistogramSelector<IDepthHistogram> selector = new DepthHistogramSelector();
+            IDepthHistogram histogram = selector.GetHistogramDepthOfTypeOrAlt(
+                new HistogramMinDepth(tableName, attributeName, Depth),
+                await DataGatherer.GetAttributeType(tableName, attributeName),
+                tableName,
+                attributeName,
+                Depth);
+            histogram.GenerateHistogramFromSortedGroups(await DataGatherer.GetSortedGroupsFromDb(tableName, attributeName));
+            return histogram;
+        }
+
+        internal async Task<IHistogram?> GetHistogramFromCacheOrNull(string tableName, string attributeName)
+        {
+            if (HistogramCacher.Instance == null)
+                return null;
+            string hash = await DataGatherer.GetTableAttributeColumnHash(tableName, attributeName);
+            var retVal = HistogramCacher.Instance.GetValueOrNull(new string[] { tableName, attributeName, hash, Depth.ToString() });
```

Comment:
Missing `typeof(HistogramMinDepth).Name` as well (Look at the `CacheHistogram()` function to see what strings are used in the hash)